pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/JSingleChildNode.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.ast;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * A node that has a single type of children, but can have no children.
+ *
+ * Package private, the methods are exposed nevertheless.
+ *
+ * @author Clément Fournier
+ */
+interface JSingleChildNode<T extends JavaNode> extends JavaNode {
```

Comment:
Is it important, that the interface starts with the prefix "J"? We don't usually do this...

Reply:
I only did this, because this interface is equivalent to one that is introduced in #1622, and I want to avoid the name clash. When that PR is merged we can remove this one.